### ISSUE TITLE
Kiss FC  ImmersionRC Ghost Protocol

### ISSUE THREAD

**Issue by artusmuc** [fedorcomander/kiss-gui#28]
As is can read at the homepage of ImmersionRC Ghost was not supported by KISS I am a "Ghost" owner - and i grabbed through the internet and found a YT video where "flyduiono kisses a ImmersionRC Ghost" :)

Is there any possibilty to have such a firmware "soon" ? I am planning now my 5 Inch Kiss Quad - and i dont want to use a reciever with inverted s-bus.

This is not an issue - only a kind question :)

**Comment by wackfpv** [comment]
As of right now the firmware that supports the ghost protocol is in alpha and only available through Alex's Patreon. The kiss gui just requires a simple hack to activate the super secret code!

**Comment by fedorcomander** [comment]
awailable for my patreons for 2 month now. 

---
Cheers,
Alex

**Comment by fedorcomander** [comment]
i did donated the code to fettec. could be released soon. no idea about fettec release pipeline though. 

---
Cheers,
Alex

**Comment by artusmuc** [comment]
as i understand even if i make donation to alex via patreon i was not able to use ghost protocol on a fettec fc's? 

i am not familiar with patreon. i try to find out since 1 hour how to support you. Is there any link where i can follow?

**Comment by fedorcomander** [comment]
First its not a donation. Its a support pledge.

Second - yes, it will work on fettec ;)

**Comment by artusmuc** [comment]
Yes .. i payed this support pledged.. and i have some questions later .. your "secret" is not working for MacOS ..
I am a working nor a video how difficult it is to make a 55 Euro FC an an 70 Euro ESC to geht working with an non standard RC system like GHOST and especially as MacUser 
Dont be surpiced that my Optionen ist .. "making $$" 
.. .and i will "donate" my experice to imercionRc also

**Comment by wackfpv** [comment]
What gui version are you using? The hack only works with v2.0.27

**Comment by fedorcomander** [comment]
1 please stop, its for alfa testers. Doesn't mean to be “final” solution. It works now, u can test. i DONT do GUI - and everything related to gui needs to go via fettec, which is pita.
2 i gave ghost code like month ago to master/fettec. That will be public. Easy peasy.
3 There are 3!!! ghost users i am know of… If u dont want hustle - use other protocol.

**Comment by artusmuc** [comment]
So .. i have now looked around some hours let's start a new round

Setup... i have a KissCC 
I want to connect a GPS - a miniOSD and a Ghost/Trampnano combination
I have installed mwosd 1.9.2.0 on the minimOSD with "kiss & gps" option

OSD is connected to TX/RX 3
minimOSD is connected to TX/RX1 

Reciever is still not connected ... 

With my macOS .. the "secret workaround" does not work - cause there is no configuration file.
Ok .. i am using my Win7 installation on my mac ..and yes i was on GUI 2.0.29

I will make a quick test with ..0.27.

With powering up via usb .. GPS is working - seeing data in GUI 
With powering up via battery and connecting via usb i have grayed out "advanced page" from 
Custom FC orientation to VTX configuration .. 

But ok .. i will download ....27 and give a try ..

**Comment by artusmuc** [comment]
And sorry .. i was a litte frustated yesterday after spending hours .. and today i try to find whats going wrong .. and the only what i can see ... something changed after connecting the battery - i have now flashed now the firmware .. i dont know - 20 times between 1.3 available via remote and ..45Y now available ...

**Comment by artusmuc** [comment]
So .. a new test with a succes

I have connected my flyduino v.1.03 FC .. with GPS & OSD 

Tada ... it works .. i have OSD & GPS connected 
To RX1/TX2 and rx2/TX2
Flashed with .... 45 firmware and controllled with 2.0.29 GUI 
MinimODS 1.9.2.0 with all Kiss "opens" i fount 

Shows me .. all what i want for have GPS works fine and OSD works fine 

Is there a way to find out why its not possible with the CC ?

**Comment by artusmuc** [comment]
..... done

**Comment by gpomykala** [comment]
I checked that IRC Ghost support was added in 1.3RC46z FETTEC KISS fw. 

 Can someone explain / point me how to solder / configure ghost receiver with KISS/ Fettec F7? So far i only used Betaflight FCs and I'm a bit lost
